Poly/UvPoly/Basic.lean

```diff
@@ -333,86 +334,13 @@ theorem proj_snd {Γ X : C} {P : UvPoly E B} {f : Γ ⟶ P @ X} :
-  toFun := P.proj
-  invFun u := P.lift (Γ := Γ) (X := X) u.1 u.2
-  left_inv f := by
-    dsimp
-    symm
-    fapply partialProd.hom_ext ⟨fan P X, isLimitFan P X⟩
-    · simp [partialProd.lift]
-      rfl
-    · sorry
-  right_inv := by
-    intro ⟨b, e⟩
-    ext
-    · simp only [proj_fst, lift_fst]
-    · sorry
-
-#exit
-
-/-- `UvPoly.equiv` is natural in `Γ`. -/
-lemma equiv_naturality_left {Δ Γ : C} (σ : Δ ⟶ Γ) (P : UvPoly E B) (X : C)
-    (f : Γ ⟶ P @ X) :
-    equiv P Δ X (σ ≫ f) = let ⟨b, e⟩ := equiv P Γ X f
-                           ⟨σ ≫ b, pullback.lift (pullback.fst .. ≫ σ) (pullback.snd ..)
-                                     (assoc (obj := C) .. ▸ pullback.condition) ≫ e⟩ := by
-  dsimp
-  congr! with h
-  . simp [polyPair, partialProduct.liftAux]
-  . set g := _ ≫ (P.polyPair be).snd
-    rw [(_ : (P.polyPair (σ ≫ be)).snd = (pullback.congrHom h rfl).hom ≫ g)]
-    · generalize (P.polyPair (σ ≫ be)).fst = x at h
-      cases h
-      simp
-    · simp only [polyPair, comp_obj, homEquiv_counit, id_obj, comp_left, pullback_obj_left,
-      mk_left, mk_hom, star_obj_left, pullback_map_left, homMk_left, pullback.congrHom_hom, ←
-      assoc, g]
-      congr 2
-      ext <;> simp
-
-/-- `UvPoly.equiv` is natural in `X`. -/
-lemma equiv_naturality_right {Γ X Y : C}
-    (P : UvPoly E B) (be : Γ ⟶ P.functor.obj X) (f : X ⟶ Y) :
-    equiv P Γ Y (be ≫ P.functor.map f) =
-      let ⟨b, e⟩ := equiv P Γ X be
-      ⟨b, e ≫ f⟩ := by
-  dsimp
-  congr! 1 with h
-  . simp [polyPair]
-  . set g := (P.polyPair be).snd ≫ f
-    rw [(_ : (P.polyPair (be ≫ P.functor.map f)).snd = (pullback.congrHom h rfl).hom ≫ g)]
-    · generalize (P.polyPair (be ≫ P.functor.map f)).fst = x at h
-      cases h
-      simp
-    · dsimp only [polyPair, g]
-      rw [homMk_comp (w_f := by simp [fstProj, functor]) (w_g := by simp [functor])]
-      simp only [UvPoly.functor, Functor.comp_map, forget_map, homMk_eta,
-        homEquiv_naturality_right_symm, comp_left, assoc]
-      admit
-      --rw [show ((Over.pullback E).map f).left ≫ prod.snd = prod.snd ≫ f by simp]
-      -- simp only [← assoc]
-      -- congr 2
-      -- simp only [comp_obj, forget_obj, star_obj_left, homEquiv_counit, id_obj, comp_left,
-      --   pullback_obj_left, mk_left, mk_hom, pullback_map_left, Over.homMk_left,
-      --   pullback.congrHom_hom, ← assoc]
-      -- congr 1
-      -- ext <;> simp
-
 /-- The domain of the composition of two polynomials. See `UvPoly.comp`. -/
 def compDom {E B D A : C} (P : UvPoly E B) (Q : UvPoly D A) :=
   Limits.pullback Q.p (fan P A).snd
 
-/-- The codomain of the composition of two polynomials. See `UvPoly.comp`. -/
-def compCod {E B D A : C} (P : UvPoly E B) (_ : UvPoly D A) :=
-  P @ A
-
 @[simps!]
 def comp [HasPullbacks C] [HasTerminal C]
-    {E B D A : C} (P : UvPoly E B) (Q : UvPoly D A) : UvPoly (compDom P Q) (compCod P Q) :=
+    {E B D A : C} (P : UvPoly E B) (Q : UvPoly D A) : UvPoly (compDom P Q) (P @ A) :=
```

Comment:
`compDom` seems useful to have as an abbreviation that gets referred to a lot, but for the codomain I think just `P @ A` is ok; more definitions generally entail needing more lemmas later.